src/m365/spo/commands/page/page-control-set.ts

```diff
@@ -0,0 +1,177 @@
+            if (this.verbose) {
+              logger.log('webPartProperties data:');
+              logger.log(args.options.webPartProperties);
+              logger.log('');
+            }
+        
+            const webpartProperties = JSON.parse(args.options.webPartProperties);
+        
+            csControl.webPartData.properties = {
+              ...csControl.webPartData.properties,
+              ...webpartProperties
+            };        
+        
+            if (this.verbose) {
+              logger.log('Updated webpart properties:');
+              logger.log(csControl.webPartData.properties);
+              logger.log('');
+            }
+          }
+        
+          return Page.save(args.options.name, args.options.webUrl, canvasContent, logger, this.debug, this.verbose);
+        } else {
+          return Promise.reject(`Control with ID ${args.options.id} not found on page ${args.options.name}`);
+        }
+      })
+      .then(() => {
+        if (this.verbose) {
+          logger.log(chalk.green('DONE'));
+        }
+        cb();
+      })
+      .catch((err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --id <id>',
+        description: 'ID of the control to set new properties for'
+      },
+      {
+        option: '-n, --name <name>',
+        description: 'Name of the page where the control is located'
+      },
+      {
+        option: '-u, --webUrl <webUrl>',
+        description: 'URL of the site where the page to retrieve is located'
+      },
+      {
+        option: '--webPartData [webPartData]',
+        description: 'JSON string with web part data as retrieved from the web part maintenance mode. Specify webPartProperties or webPartData but not both'
+      },
+      {
+        option: '--webPartProperties [webPartProperties]',
+        description: 'JSON string with web part data as retrieved from the web part maintenance mode. Specify webPartProperties or webPartData but not both'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!Utils.isValidGuid(args.options.id)) {
+      return `${args.options.id} is not a valid GUID`;
+    }
+
+    if (args.options.webPartProperties) {
+      try {
+        JSON.parse(args.options.webPartProperties);
+      } catch (e) {
+        return `Specified webPartProperties is not a valid JSON string. Input: ${args.options.webPartData}. Error: ${e}`;
+      }
+    }
+
+    if (args.options.webPartData) {
+      try {
+        JSON.parse(args.options.webPartData);
+      } catch (e) {
+        return `Specified webPartData is not a valid JSON string. Input: ${args.options.webPartData}. Error: ${e}`;
+      }
+    }
+
+    if (args.options.webPartData && args.options.webPartProperties) {
```

Comment:
Let's move this check up so that we don't unnecessarily parse JSON if both arguments have been provided and the command will fail anyway.